packages/web/src/components/layouts/SearchResults/SearchResults.tsx

```diff
@@ -0,0 +1,337 @@
+        ],
+        numberOfUses: 5265,
+      },
+    },
+    {
+      id: 'TYIwvAfy',
+      title: 'react-smooth',
+      description: 'is a animation library work on React',
+      featuredSites: {
+        iconList: [
+          'https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg',
+          'https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg',
+          'https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg',
+        ],
+        numberOfUses: 5265,
+      },
+    },
+    {
+      id: 'Lq1pEEX7',
+      title: 'unist-util-position',
+      description: 'unist utility to get the positional info of nodes',
+      featuredSites: {
+        iconList: [
+          'https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg',
+          'https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg',
+          'https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg',
+        ],
+        numberOfUses: 5265,
+      },
+    },
+    {
+      id: 'cWOgIbmp',
+      title: 'vfile-message',
+      description: 'Create vfile messages',
+      featuredSites: {
+        iconList: [
+          'https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg',
+          'https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg',
+          'https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg',
+        ],
+        numberOfUses: 5265,
+      },
+    },
+    {
+      id: 'UT97Vpoi',
+      title: 'Go to all Popular packages',
+      variant: 'toAll',
+    },
+  ];
+
+  const keyWords = {
+    fullList: [
+      {
+        id: '#art',
+        name: '#art',
+      },
+      {
+        id: '#angular',
+        name: '#angular',
+      },
+      {
+        id: '#moment',
+        name: '#moment',
+      },
+      {
+        id: '#date',
+        name: '#date',
+      },
+      {
+        id: '#react',
+        name: '#react',
+      },
+      {
+        id: '#parse',
+        name: '#parse',
+      },
+      {
+        id: '#fb',
+        name: '#fb',
+      },
+    ],
+    featuredItems: ['#moment', '#date', '#react', '#parse', '#fb'],
+  };
+
+  const vulnerabilities = ['Vulnerabilities', 'Outdated', 'Duplicate'];
+
+  const authors = {
+    fullList: [
+      {
+        id: 'acdlite',
+        name: 'acdlite',
+      },
+      {
+        id: 'gaearon',
+        name: 'gaearon',
+      },
+      {
+        id: 'sophiebits',
+        name: 'sophiebits',
+      },
+      {
+        id: 'trueadm',
+        name: 'trueadm',
+      },
+    ],
+    featuredItems: ['acdlite', 'gaearon', 'sophiebits', 'trueadm'],
+  };
+
+  const [selectedKeywords, setSelectedKeywords] = useState<string[] | []>([]);
+  const [selectedProblems, setSelectedProblems] = useState<string[] | []>([]);
+  const [selectedAuthors, setSelectedAuthors] = useState<string[] | []>([]);
```

Comment:
Just a note: in general, you don't need to specify an empty array type explicitly, `[]` is perfectly assignable to `string[]`. 

Anyway, probably I could live with this _here_, because filters will be managed by global app state manager afterwards.